src/main/java/Duke.java

```diff
@@ -1,10 +1,266 @@
+        int number = Integer.valueOf(message[1]);
+        Task task = list.get(number-1);
+        list.remove(number-1);
+        return task;
+    }
+
+    private static Todo createTodo(String input,boolean isDone) throws DukeTodoException{
+
+        int startOfMessage = 5;
+        int endOfMessage = input.length();
+        if(endOfMessage <= startOfMessage){
+            throw new DukeTodoException();
+        }
+        String message = input.substring(startOfMessage,endOfMessage);
+        if(message.isEmpty()){
+            throw new DukeTodoException();
+        }else {
+            Todo temp = new Todo(message, isDone);
+            return temp;
+        }
+    }
+
+    private static Deadline createDeadline(String input,boolean isDone) throws DukeDeadlineException{
+        int startOfMessage = 9;
+        int endOfMessage = input.indexOf("/by")-1;
+        int startOfBy = input.indexOf("/by") + 4;
+        int endOfBy = input.length();
+        if(endOfMessage <= startOfMessage) {
+            throw new DukeDeadlineException();
+        }
+        String message = input.substring(startOfMessage, endOfMessage);
+        String by = input.substring(startOfBy,endOfBy);
+        if(message.isEmpty() || by.isEmpty()){
+            throw new DukeDeadlineException();
+        }else {
+            Deadline temp = new Deadline(message, isDone, by);
+            return temp;
+        }
+    }
+
+    private static Event createEvent(String input,boolean isDone) throws DukeEventException{
+        int startOfMessage = 6;
+        int endOfMessage = input.indexOf("/at")-1;
+        int startOfAt = input.indexOf("/at") + 4;
+        int endOfAt = input.length();
+        if(endOfMessage <= startOfMessage) {
+            throw new DukeEventException();
+        }
+        String message = input.substring(startOfMessage, endOfMessage);
+        String at = input.substring(startOfAt,endOfAt);
+        if(message.isEmpty() || at.isEmpty()){
+            throw new DukeEventException();
+        }
+        Event temp = new Event(message, isDone,at);
+        return temp;
+    }
+
+    private static void saveTask(List<Task> list) throws IOException{
+        FileWriter fileWriter = new FileWriter(FILENAME);
+        for(Task task : list){
+            if(task instanceof Todo){
+                fileWriter.write("T | " + (task.isDone ? "1" : "0") + " | " +
+                        task.description + "\n");
+            }else if(task instanceof Deadline){
+                String[] deadline =  task.description.split("/by");
+                fileWriter.write("D | " + (task.isDone ? "1" : "0") + " | " +
+                        task.description + " | " + ((Deadline) task).by + "\n");
+            }else if(task instanceof Event) {
+                fileWriter.write("E | " + (task.isDone ? "1" : "0") + " | " +
+                        task.description + " | " + ((Event) task).at + "\n");
+            }
+        }
+        fileWriter.close();
     }
 }
```

Comment:
Just a suggestion: the Duke java file is probably too long for a single file. Maybe a "FileManager", a "PrintManager", or a "TaskManager" can help reduce the length of this file.